locator/topology.cc

```diff
@@ -198,16 +198,16 @@ const node* topology::add_node(node_holder nptr) {
 
         tlogger.debug("topology[{}]: add_node: {}, at {}", fmt::ptr(this), node_printer(nptr.get()), lazy_backtrace());
 
-        index_node(node);
+        index_node(*node);
     } catch (...) {
-        pop_node(make_mutable(node));
+        pop_node(*node);
         throw;
     }
-    return node;
+    return *node;
 }
 
-const node* topology::update_node(node* node, std::optional<host_id> opt_id, std::optional<inet_address> opt_ep, std::optional<endpoint_dc_rack> opt_dr, std::optional<node::state> opt_st, std::optional<shard_id> opt_shard_count) {
-    tlogger.debug("topology[{}]: update_node: {}: to: host_id={} endpoint={} dc={} rack={} state={} shard_count={}, at {}", fmt::ptr(this), node_printer(node),
+const node& topology::update_node(const node& node, std::optional<host_id> opt_id, std::optional<inet_address> opt_ep, std::optional<endpoint_dc_rack> opt_dr, std::optional<node::state> opt_st, std::optional<shard_id> opt_shard_count) {
+    tlogger.debug("topology[{}]: update_node: {}: to: host_id={} endpoint={} dc={} rack={} state={} shard_count={}, at {}", fmt::ptr(this), node_printer(&node),
```

Comment:
Was this `const` simply incorrectly  missing, or am I missing something? I see `auto mutable_node = make_mutable(node);` in the old code below.

Reply:
Also, it seems weird that we pass a const reference and return a const reference when the returned reference points to the same object as the passed reference. Can't we just pass a non-const reference and return `void`? It would be more clear what this function does.

`topology::update_node` seems to be used only in `topology::add_or_update_endpoint` and `locator_topology_test.cc`, so only a small refactoring should be required.

Reply:
Yes it makes more sense to refactor the `update_node` to take in `node&` and return `void`

Reply:
done @patjed41